moped-editor/src/views/projects/projectView/ProjectMilestones.js

```diff
@@ -28,14 +27,167 @@ import {
 import { useMutation } from "@apollo/client";
 import { format } from "date-fns";
 import parseISO from "date-fns/parseISO";
-import Autocomplete from "@mui/material/Autocomplete";
 
-// Helpers
 import { phaseNameLookup } from "src/utils/timelineTableHelpers";
-import DateFieldEditComponent from "./DateFieldEditComponent";
 import ToggleEditComponent from "./ToggleEditComponent";
-import ButtonDropdownMenu from "../../../components/ButtonDropdownMenu";
-import MilestoneTemplateModal from "./MilestoneTemplateModal";
+import MilestoneTemplateModal from "./ProjectMilestones/MilestoneTemplateModal";
+import MilestoneAutocompleteComponent from "./ProjectMilestones/MilestoneAutocompleteComponent";
+import DataGridDateFieldEdit from "./ProjectMilestones/DataGridDateFieldEdit";
+import DeleteConfirmationModal from "./DeleteConfirmationModal";
+
+const useMilestoneNameLookup = (data) =>
+  useMemo(() => {
+    if (!data) {
+      return {};
+    }
+    return data.moped_milestones.reduce(
+      (obj, item) =>
+        Object.assign(obj, {
+          [item.milestone_id]: item.milestone_name,
+        }),
+      {}
+    );
+  }, [data]);
+
+const useColumns = ({
+  data,
+  rowModesModel,
+  handleEditClick,
+  handleSaveClick,
+  handleCancelClick,
+  handleDeleteOpen,
+  milestoneNameLookup,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "Milestone",
+        field: "milestone_id",
+        renderCell: ({ row }) => row.moped_milestone?.milestone_name,
+        // input validation:
+        preProcessEditCellProps: (params) => ({
+          ...params.props,
+          error: !params.props.value,
+        }),
+        editable: true,
+        renderEditCell: (props) => (
+          <MilestoneAutocompleteComponent
+            {...props}
+            milestoneNameLookup={milestoneNameLookup}
+          />
+        ),
+        width: 250,
+      },
+      {
+        headerName: "Description",
+        field: "description",
+        width: 200,
+        editable: true,
+        renderEditCell: (props) => <DataGridTextField {...props} />,
+      },
+      {
+        headerName: "Related phase",
+        field: "moped_milestone",
+        editable: false, // would it be cool for this to update when someone edits the milestone
+        valueGetter: (value) => {
+          return phaseNameLookup(data)[value?.related_phase_id] ?? "";
+        },
+        width: 150,
+      },
+      {
+        headerName: "Completion estimate",
+        field: "date_estimate",
+        editable: true,
+        valueFormatter: (value) =>
+          value ? format(parseISO(value), "MM/dd/yyyy") : undefined,
+        renderEditCell: (props) => (
+          <DataGridDateFieldEdit
+            {...props}
+            name="date_estimate"
+            label="Completion estimate"
+          />
+        ),
+        width: 175,
+      },
+      {
+        headerName: "Date completed",
```

Comment:
@chiaberry Could you add some space between the completion estimate and date completed fields? I think that we ran into the same thing in the funding table when migrating and went with `theme.spacing(1)` iirc.

<img width="397" alt="Screenshot 2024-10-29 at 10 30 00 AM" src="https://github.com/user-attachments/assets/ab6af1a7-78fe-4fba-958b-cefe482f53f4">

Reply:
done!